packages/compass-shell/src/modules/worker-runtime/worker.js

```diff
@@ -0,0 +1,64 @@
+import { parentPort } from 'worker_threads';
+import { ElectronRuntime } from '@mongosh/browser-runtime-electron';
+import { CliServiceProvider } from '@mongosh/service-provider-server';
+import { exposeAll, createCaller } from './rpc';
+import { serializeResult, serializeError } from './worker-shell-result';
+
+let provider = null;
+let runtime = null;
+
+const evaluationListener = createCaller(
+  ['onPrint', 'onPrompt', 'toggleTelemetry'],
+  parentPort
+);
+
+async function init(uri, options, cliOptions) {
+  provider = await CliServiceProvider.connect(uri, options, cliOptions);
+  runtime = new ElectronRuntime(provider /** , TODO: messageBus? */);
```

Comment:
Yeah, I think we’d need the messageBus for telemetry? But anything that vaguely looks like an `EventEmitter` should suffice here